### PR TITLE
Issue 324 metadata is str after saving empty dict

### PR DESCRIPTION
**Pull request by nikolamaric**

### CODE REVIEW

File: solvebio/resource/apiresource.py [hunk 1]
```diff
@@ -326,7 +326,8 @@ def serialize(self, obj):
 
             if isinstance(attr, SolveObject):
                 params[k] = dict(attr)
-            else:
-                params[k] = attr or ""
+                continue
```
Comment (davecap) [inline solvebio/resource/apiresource.py:329]: is this for testing?

Comment (nikolamaric) [reply solvebio/resource/apiresource.py:329]: How do you mean for testing?

Comment (nikolamaric) [reply solvebio/resource/apiresource.py:329]: the latest changes are:
https://github.com/solvebio/solvebio-python/blob/2d71c5a0d91a98ccfa10ebcc556b782d12b303ef/solvebio/resource/apiresource.py#L327

Comment (davecap) [reply solvebio/resource/apiresource.py:329]: Oh I misread the code. Is the continue necessary? Why not an if/else

Comment (nikolamaric) [reply solvebio/resource/apiresource.py:329]: In this case, `continue `does the same as `else`, however, I can change it to be if/else.

Comment (davecap) [reply solvebio/resource/apiresource.py]: yes definitely, just feel that if/else is easier to follow

File: solvebio/test/test_object.py [hunk 2]
```diff
@@ -257,3 +258,22 @@ def test_object_set_metadata(self):
...
+            get_or_create_by_full_path('~/{}'.format(uuid.uuid4()), object_type='folder')
+
+        folder.metadata['foo'] = 'bar'
+        folder.save()
+
+        folder.metadata = {}
+        folder.save()
+
+        # Test that 'metadata' is an empty dict
+        self.assertTrue(isinstance(folder.metadata, dict) and not folder.metadata)
```
Comment (davecap) [inline solvebio/test/test_object.py:273]: why not just assert metadata == {}?

Comment (nikolamaric) [reply solvebio/test/test_object.py]: My bad. Done.

File: solvebio/test/test_object.py [hunk 3]
```diff
@@ -257,3 +258,22 @@ def test_object_set_metadata(self):
...
+        folder.save()
+
+        # Test that 'metadata' is an empty dict
+        self.assertTrue(folder.metadata == {})
+
+        folder.metadata = []
+        folder.save()
+
+        # Test that 'metadata' is an empty list
+        self.assertTrue(folder.metadata == [])
```
Comment (jsh2134) [inline solvebio/test/test_object.py:279]: can you delete the folder at the end of the test? `folder.delete(force=True)` (and I think in other test too)

Comment (nikolamaric) [reply solvebio/test/test_object.py:281]: Done.

**Comment by nikolamaric** [comment]
closes #324

**Comment by jsh2134** [comment]
closes #324

**Pull request merged**